tests/Data/UuidTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+require_once("libs/composer/vendor/autoload.php");
+
+use PHPUnit\Framework\TestCase;
+use ILIAS\Data\UUID\Factory;
+use ILIAS\Data\UUID\RamseyUuidWrapper;
+use Ramsey\Uuid\Exception\InvalidUuidStringException;
+
+class UuidTest extends TestCase
+{
+    const VALID_UUID4 = '/^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[0-9a-f]{4}-[0-9a-f]{12}$/';
+
+    const UUID4 = 'f47ac10b-58cc-4372-a567-0e02b2c3d479';
+    const NO_UUID = 'lorem ipsum dolor';
+
+    /**
+     * @doesNotPerformAssertions
+     */
+    public function test_init()
+    {
+        return new Factory();
+    }
+
+    /**
+     * @depends test_init
+     */
+    public function test_uuid4()
+    {
+        $factory = new Factory();
+        $uuid = $factory->uuid4();
+
+        $this->assertTrue(get_class($uuid) === RamseyUuidWrapper::class);
```

Comment:
```suggestion
```

We don't need to test this. The fact that this is indeed a `Uuid` is already guaranteed via the return type hint on the factory (right?). The fact, that this is a `RamseyUuidWrapper` is an implementation detail we don't need to test IMO.